src/codegate/storage/storage_engine.py

```diff
@@ -152,66 +130,99 @@ async def search(
-                        wvc.query.Filter.by_property("name").contains_any(packages),
-                        wvc.query.Filter.by_property("type").equal(ecosystem)
-                    ]),
+                placeholders = ",".join("?" * len(packages))
+                query_sql = f"""
+                    SELECT name, type, status, description
+                    FROM packages
+                    WHERE LOWER(name) IN ({placeholders})
+                    AND LOWER(type) = ?
+                """  # nosec
+                params = [p.lower() for p in packages] + [ecosystem.lower()]
+                logger.debug(
+                    "Searching by package names and ecosystem",
+                    packages=packages,
+                    ecosystem=ecosystem,
+                    sql=query_sql,
+                    params=params,
                 )
-                response.objects = [
-                    obj
-                    for obj in response.objects
-                    if obj.properties["name"].lower() in packages
-                    and obj.properties["type"].lower() == ecosystem.lower()
-                ]
+                cursor.execute(query_sql, params)
+
             elif packages and not ecosystem:
-                response = collection.query.fetch_objects(
-                    filters=wvc.query.Filter.all_of([
-                        wvc.query.Filter.by_property("name").contains_any(packages),
-                    ]),
+                placeholders = ",".join("?" * len(packages))
+                query_sql = f"""
+                    SELECT name, type, status, description
+                    FROM packages
+                    WHERE LOWER(name) IN ({placeholders})
+                """  # nosec
+                params = [p.lower() for p in packages]
+                logger.debug(
+                    "Searching by package names only",
+                    packages=packages,
+                    sql=query_sql,
+                    params=params,
                 )
-                response.objects = [
-                    obj
-                    for obj in response.objects
-                    if obj.properties["name"].lower() in packages
-                ]
+                cursor.execute(query_sql, params)
+
             elif query:
-                # Perform the vector search
-                # Generate the vector for the query
+                # Generate embedding for the query
                 query_vector = await self.inference_engine.embed(self.model_path, [query])
-
-                response = collection.query.near_vector(
-                    query_vector[0],
+                query_embedding = np.array(query_vector[0], dtype=np.float32)
+                query_embedding_bytes = query_embedding.tobytes()
+
+                query_sql = """
+                    WITH distances AS (
+                        SELECT name, type, status, description,
+                               vss_distance(embedding, ?) as distance
```

Comment:
We need to use cosine distance here. I could not find documentation on vss_distance function, and not sure which kind of distance it computes. I see a vec_distance_cosine function which we can use instead.

https://alexgarcia.xyz/sqlite-vec/api-reference.html#vec_distance_cosine

Reply:
Thanks ! well captured , as per slack:  I started out with sqlite-vss and then got a warning it was deprecated, so flipped to the sqlite-vec, but left the old similarity distance call in place which weirdly still worked